contracts/PathofTheTemplarShard.sol

```diff
@@ -0,0 +1,106 @@
+pragma solidity ^0.8.13;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "./PartnerMinter.sol/";
+
+interface IRelic {
+    function balanceOf(address) external returns (uint256);
+    function tokenOfOwnerByIndex(address, uint256) external returns (uint256);
+    function getRelicInfos(uint256 enclaves) external returns (uint256);
+}
+
+interface IShards {
+    function partnerMint(
+        address account,
+        uint256 id,
+        uint256 amount,
+        bytes memory data
+    ) external;
+}
+
+contract PathofTheTemplarShard is ownable {
+
+    event SignedQuestCompletedMessageHash(bytes32 hash);
+
+    IShards private SHARDS,
+    IRelic private RELIC,
+    uint256[] public SHARD_ID = [1, 2, 3, 4, 5];
+
+mapping (msg.sender => bool) public canMint
+mapping (uint => enclaves) signedQuestCompletedMessage
+
+function mintPathofthetemplarShard() external canMint {
+    SHARDS.partnerMint(msg.sender, SHARD_ID, 1, "");
+    require(msg.sender(signedQuestCompletedMessage)) == "true";
+}
+
+function signedQuestCompletedMessage(QuestCompletedMessageReq calldata req, bytes calldata signature) external {
+
+        bytes32 digest = keccak256(abi.encodePacked(
+            "\x19\x01",
+            DOMAIN_SEPARATOR,
+            hash(req)
+        ));
+
+        (address signer, ECDSA.RecoverError err) = ECDSA.tryRecover(digest, signature);
+        if (err != ECDSA.RecoverError.NoError) {
+            revert InvalidSignature(req.account);
+        }
+        if (block.timestamp > req.deadline) revert DeadlineExpired(block.timestamp - req.deadline);
+        if (signer != req.account) revert InvalidSignature(req.account);
+        if (_useNonce(req.account) != req.nonce) revert InvalidNonce(req.account);
+
+        _setQuestCompletedMessage(req.account, req.discordIds);
+    }
+
+    /**
+     * "Consume a nonce": return the current value and increment.
+     */
+    function _useNonce(address _owner) internal returns (uint256 current) {
+        Counters.Counter storage nonce = nonces[_owner];
+        current = nonce.current();
+        nonce.increment();
+    }
+
+    struct EIP712Domain {
+        string name;
+        string version;
+        uint256 chainId;
+    }
+
+    bytes32 constant EIP712DOMAIN_TYPEHASH = keccak256("EIP712Domain(string name,string version,uint256 chainId)");
+
+    function hash(EIP712Domain memory _input) internal pure returns (bytes32) {    
+        return keccak256(abi.encode(
+            EIP712DOMAIN_TYPEHASH,
+            keccak256(bytes(_input.name)),
+            keccak256(bytes(_input.version)),
+            _input.chainId
+            emit SignedQuestCompletedMessageHash(hash);
+        ));
+    }
+
+    struct QuestCompletedMessageReq {
+        address account;
+        uint256[] discordIds;
```

Comment:
What's the intent for this to work end-to-end? Is `account` and `discordIds` still relevant for your use case?

Reply:
We still have discordIds from the firebase service account that gets read by the server which tracks the user progression through the ACTS in the quest. However, it probably isn't necessary since we are not interfacing from the Templar.sol or TemplarMetadata.sol. It is also not needed as the information from firebase does not get communicated to the contract as a dependency for minting. 

If we remove `address account;`, would the contract still be able to parse who the owner of the signer from ECDSA is?